### ISSUE TITLE
[BUG] mx.random.normal can produce -inf

### ISSUE THREAD

**Issue by davidkoski** [ml-explore/mlx#642]
**Describe the bug**
mx.random.normal can produce -inf

**To Reproduce**

Include code snippet
```python
import mlx.core as mx

mx.random.seed(61635)
a = mx.random.normal([4, 3], dtype=mx.float16)

print(a)
```

Output is:

```
array([[-0.967285, -0.724121, 0.654785],
       [1.31836, 1.0498, -0.641113],
       [-0.123352, 0.152344, 0.524902],
       [-inf, 0.279053, -1.46973]], dtype=float16)
```

Notice the item in the lower left.

**Expected behavior**

Expected: no `-inf` in the result.

**Desktop (please complete the following information):**
 - OS Version: 14.2.1
 - Version 16750f3c51921e0a632071ff30532d9d6afa7fb4

**Additional context**

**Comment by davidkoski** [comment]
Another seed that produces it:

```
MLXRandom.seed(53978)
```

**Comment by awni** [comment]
Should be fixed in #904
